.github/workflows/build_for_pypi.yml

```diff
@@ -1,46 +1,90 @@
+---
 name: Build and Optionally Publish to PyPi
 
 on:
-  workflow_call:
-    inputs:
-      publish:
-        type: boolean
-        default: false
-        description: "Publish to PyPi"
+  pull_request:
+    branches:
+      - main
+  push:
+    # Pattern matched against refs/tags
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
 
 jobs:
-  build_for_pypi:
+  build_dist_for_pypi:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-13, macos-latest]
```

Comment:
This should also include `ubuntu-24.04-arm` that GH recently made available (it's a public preview, but it works).

Additionally, this is lacking other architectures: the matrix needs a way to represent a need to set up QEMU, which should then have its own step.